src/ucp/core/ucp_mm.c

```diff
@@ -645,7 +650,19 @@ ucs_status_t ucp_memh_register(ucp_context_h context, ucp_mem_h memh,
                                         UCS_LOG_LEVEL_ERROR;
 
     return ucp_memh_register_internal(context, memh, md_map, uct_flags,
-                                      alloc_name, err_level, 1);
+                                      alloc_name, err_level, 1, 1);
+}
+
+void ucp_memh_disable_gva(ucp_mem_h memh, ucp_md_map_t md_map)
+{
+    ucp_context_h context = memh->context;
+    ucs_status_t UCS_V_UNUSED status;
+
+    memh->md_map &= ~context->gva_md_map[memh->mem_type];
+    status = ucp_memh_register_internal(context, memh, md_map, 0, "disable gva",
+                                        UCS_LOG_LEVEL_DIAG, 1, 0);
+    ucs_assert(status == UCS_OK);
```

Comment:
why assert?

Reply:
to emphasis that function can't fail.
currently when passing `allow_partial_reg` status may only be OK, but if in future it changes we might need do something here

Reply:
`ucs_assert_always(status == UCS_OK); /* When allow_partial_reg==1 registration should not fail */`